src/parse.rs

```diff
@@ -1,40 +1,187 @@
+use crate::cel::CelId;
 use crate::external_file::{ExternalFile, ExternalFilesById};
+use crate::layer::{LayerData, LayersData};
 use crate::reader::AseReader;
+use crate::slice::Slice;
 use crate::tileset::{Tileset, TilesetsById};
+use crate::user_data::UserData;
 use crate::{error::AsepriteParseError, AsepriteFile, PixelFormat};
 use log::debug;
 use std::io::Read;
 
 use crate::Result;
 use crate::{cel, color_profile, layer, palette, slice, tags, user_data, Tag};
 
+enum LayerParseInfo {
```

Comment:
Can you maybe add a comment what the idea is behind this type?

Reply:
Prior to this change, the layer data in the first frame was stored in the `found_layers` vec. After the first frame was done, that data moved into LayersData`, and then into `ParseInfo. I moved `found_layers` into the ParseData so we can attach UserData to it without accessing a separate variable.

I suppose we could actually remove LayersData from ParseInfo, put `layer_parents: Option<_>` on the ParseInfo directly, and set it to `Some(parents)`. For now I've left a comment explaining how it works.